kinto/core/resource/__init__.py

```diff
@@ -950,6 +950,10 @@ def _extract_filters(self, queryparams=None):
                 if has_invalid_value:
                     raise_invalid(self.request, **error_details)
 
+            if field in ('id', 'collection_id'):
```

Comment:
In the original issue, we found this bug by filtering the history with `?collection_id`.
But this actually applies to `group_id` or `record_id` too.

We could do a quick fix like this one with ` field.endswith('id')`. This will solve the history filtering problem.

But the issue is wider and applies to any field. And that's what makes it hard to fix: `?toto=123 -> {"toto": "123"} or {"toto": 123}`. I think it's more at the storage level to take care of this.

My 2 cents :)